RELEASE/scripts/autoscend/auto_zone.ash

```diff
@@ -1746,6 +1746,29 @@ boolean is_ghost_in_zone(location loc)
 	return false;
 }
 
+location[int] monster_to_location(monster target)
+{
+	location[int] retval;
+	int key = 0;
+	foreach loc in $locations[]		//check all locations in the game
+	{
+		boolean mon_in_loc = false;
+		foreach idx, mon in get_monsters(loc)
+		{
+			if(target == mon)
+			{
+				mon_in_loc = true;
+			}
+		}
+		if(mon_in_loc)
+		{
+			retval[key] = loc;
+			key++;
+		}
+	}
```

Comment:
```suggestion
	foreach loc in $locations[]		//check all locations in the game
	{
		foreach idx, mon in get_monsters(loc) if (target == mon)
		{
			retval[key++] = loc;
			break;
		}
	}
```

Reply:
But also why not return `boolean [location]` for this function?

Reply:
It just occurred to me that I was trying to handle the possibility of there being multiple copies of the exact same monster in the same location. which had caused some redundant code in there. which you cleared up.

As for why not boolean[location]
me and phil just thought it would look better if it returned a location which is enumerated via an int key rather than the location being the key. I do not mind changing it to boolean if that is what other devs prefer.

@Phillammon your thoughts?